src/lib/multi_mappings.rs

```diff
@@ -142,13 +142,12 @@ impl<T: Clone + Debug + Eq + ToString + 'static> MultiMappingStore<T> {
         if !self.contains_src(src) {
             return false;
         }
-        (*self.src.borrow().get(&src).unwrap()).len() == 1
-        && self.dst
-               .borrow()
-               .get(&self.src.borrow().get(&src).unwrap().iter().next().unwrap())
-               .unwrap()
-               .len()
-           == 1
+        let num_mapped = self.dst
+                             .borrow()
+                             .get(&self.src.borrow().get(&src).unwrap().iter().next().unwrap())
+                             .unwrap()
+                             .len();
+        (*self.src.borrow().get(&src).unwrap()).len() == 1 && num_mapped == 1
```

Comment:
Isn't this a side-effect free expression? Should it be in an `assert`?

Reply:
This fn returns a bool

Reply:
Mea culpa!